### ISSUE TITLE
x/build: investigate why vet failures on tip weren't caught by builders

### ISSUE THREAD

**Issue by dmitshur** [golang/go#37053]
`go vet std` is reporting problems and exiting with a non-zero code on latest master (see #37030):

```
src $ go version
go version devel +a864cc7560 Wed Feb 5 14:32:50 2020 +0000 darwin/amd64
src $ go vet std
# sync/atomic
sync/atomic/value.go:59:44: possible misuse of unsafe.Pointer
# runtime/internal/atomic_test
runtime/internal/atomic/atomic_test.go:94:7: possible misuse of unsafe.Pointer
# strings
strings/builder.go:30:9: possible misuse of unsafe.Pointer
# runtime
runtime/alg.go:56:22: possible misuse of unsafe.Pointer
runtime/atomic_pointer.go:63:9: possible misuse of unsafe.Pointer
runtime/cgocall.go:200:13: possible misuse of unsafe.Pointer
runtime/cgocall.go:282:16: possible misuse of unsafe.Pointer
runtime/cgocall.go:287:16: possible misuse of unsafe.Pointer
[...]
src $ echo $?
2
```

Need to investigate why the builders didn't catch this. We used to have a `misc-vet-vetall` builder but that has changed in #31916. /cc @rsc @cagedmantis @toothrot

**Comment by bcmills** [comment]
[CL 176439](https://golang.org/cl/176439) explicitly disabled the `unsafeptr` check for the standard library, while enabling all others.

(See [`cmd/go/internal/work/exec.go`](https://go-review.googlesource.com/c/go/+/176439/6/src/cmd/go/internal/work/exec.go), line 1006.)

An explicit `go vet` invocation, on the other hand, runs _all_ of the default `vet` checks.

**Comment by bcmills** [comment]
This turned out to be an undetected regression in the behavior of `go vet` itself.

When we turned down `misc-vet-vetall`, we made the _implicit_ `vet` checks for standard-library packages during `go test` more aggressive, so that they run all `vet` checks except for `unsafeptr` instead of the only the high-confidence checks.

And that's why it wasn't caught: we were testing a (slightly) different property from the one we actually wanted to hold. The testing verified that “all `vet` checks except for `unsafeptr` pass for packages in the standard library”, but the property we want is “`go vet` reports no failures for packages in the standard library”. That property only holds if `go vet` runs exactly “all `vet` checks except for `unsafeptr`”.

We didn't have a regression test for that last part, so when it regressed we also didn't detect the regression in the desired property that it implies.
